core/src/byte_array.rs

```diff
@@ -87,6 +87,13 @@ impl ByteArray {
         }
     }
 
+    pub fn iter(&self) -> ByteArrayIter {
```

Comment:
That seems useful, any chance the other `*Array` types could get iterators too?

Reply:
If I have time.